File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    answer = []

    day = 0
    tmp = 0
    
    for i, each in enumerate(progresses): 
        if day * speeds[i] + each < 100:
            if tmp != 0:
                answer.append(tmp)
            tmp = 0
            daysToComplete = (100 - (day * speeds[i] + each)) // speeds[i]

            if each + (daysToComplete * speeds[i]) < 100:
                daysToComplete += 1

            day += daysToComplete
            tmp += 1
        else:
            tmp += 1
    
    if tmp != 0:
        answer.append(tmp)

    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        Queue<int[]> q = new LinkedList<>();
        
        for (int i = 0; i < progresses.length; i++) {
            q.offer(new int[] {progresses[i], i});
        }
        
        int day = 1;
        while (!q.isEmpty()) {
            int tmpCnt = 0;
            
            while (!q.isEmpty() && q.peek()[0] + speeds[q.peek()[1]] * day >= 100) {
                q.poll();
                tmpCnt += 1;
            }            
            
            if (tmpCnt != 0) {
                answer.add(tmpCnt);
            }

            day += 1;
        }
        
        return answer.stream().mapToInt(a -> a).toArray();
    }
}